### PR TITLE
ENG-464: update versioning system

### PR DESCRIPTION
**Pull request by 3s-jorgematos**
# Description

Update versioning system using: https://github.com/semantic-release/semantic-release

### CODE REVIEW

File: .releaserc.json [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    {
+      "name": "beta",
+      "prerelease": true
+    },
+    {
+      "name": "alpha",
+      "prerelease": true
+    }
+  ],
+  "defaultBranch": "main",
```
Comment (asynctomatic) [inline .releaserc.json:16]: The default branch of this repo is `master`

File: package.json [hunk 2]
```diff
@@ -0,0 +1,25 @@
+{
+  "name": "starkexpress-dotnet-client",
+  "version": "0.0.0",
+  "description": "[gha]: https://github.com/threesigmaxyz/starkexpress-dotnet-client/actions [gha-badge]: https://github.com/threesigmaxyz/starkexpress-dotnet-client/actions/workflows/main.yml/badge.svg [dotnet]: https://github.com/dotnet/core [dotnet-badge]: https://img.shields.io/static/v1?label=Built%20with&message=.NET&color=blueviolet [license]: https://opensource.org/licenses/MIT [license-badge]: https://img.shields.io/badge/License-MIT-blue.svg",
```
Comment (asynctomatic) [inline package.json:4]: I understand this is acting as a placeholder but can we change it to a more descriptive value?

File: .github/workflows/pull_request.yml [hunk 3]
```diff
@@ -1,11 +1,16 @@
 name: Pull request workflow
 
-on: pull_request
+on: push
```
Comment (asynctomatic) [inline .github/workflows/pull_request.yml:3]: Should we rename this workflow file from `pull_request.yml` now that the trigger changed?

**Review by asynctomatic — CHANGES_REQUESTED** [review]

File: .releaserc.json (hunk 1 above)
Comment (3s-jorgematos) [reply .releaserc.json]: This prop is not necessary. It was just a test and I forgot to remove it. The relevant prop is "branches".

Removed

File: package.json (hunk 2 above)
Comment (3s-jorgematos) [reply package.json]: Changed

**Review by JoseAfonsoThreeSigma — APPROVED** [review]

**Pull request merged**